dask/bag/core.py

```diff
@@ -1117,23 +1118,40 @@ def repartition(self, npartitions):
         --------
         >>> b.repartition(5)  # set to have 5 partitions  # doctest: +SKIP
         """
-        if npartitions > self.npartitions:
-            msg = ("Repartition only supports going to fewer partitions\n"
-                   " old: %d  new: %d")
-            raise NotImplementedError(msg % (self.npartitions, npartitions))
-        npartitions_ratio = self.npartitions / npartitions
-        new_partitions_boundaries = [int(old_partition_index * npartitions_ratio)
-                                     for old_partition_index in range(npartitions + 1)]
         new_name = 'repartition-%d-%s' % (npartitions, tokenize(self))
+        if npartitions == self.npartitions:
+            return self
+        elif npartitions < self.npartitions:
+            ratio = self.npartitions / npartitions
+            new_partitions_boundaries = [int(old_partition_index * ratio)
+                                         for old_partition_index in range(npartitions + 1)]
+
+            dsk = {}
+            for new_partition_index in range(npartitions):
+                value = (list, (toolz.concat,
+                                [(self.name, old_partition_index)
+                                 for old_partition_index in
+                                 range(new_partitions_boundaries[new_partition_index],
+                                       new_partitions_boundaries[new_partition_index + 1])]))
+                dsk[new_name, new_partition_index] = value
+        else:  # npartitions > self.npartitions
+            ratio = npartitions / self.npartitions
+            split_name = 'split-%s' % tokenize(self)
+            dsk = {}
+            last = 0
+            j = 0
+            for i in range(self.npartitions):
+                new = last + ratio
+                if i == self.npartitions - 1:
+                    k = npartitions - j
+                else:
+                    k = int(new) - int(last)
```

Comment:
Perhaps move these int casts to just casting on the original `ratio`, or on `new=last + ratio`?